src/cases/components/case-task/case-task.util.ts

```diff
@@ -0,0 +1,23 @@
+import { Task } from '../../../work-allocation-2/models/tasks';
+
+export const appendTaskIdAsQueryStringToTaskDescription = (task: Task): string => {
+  // The url format will be like the following markdown:
+  // 'Click link to proceed to next step [test link next step](/cases/case-details/1547652071308205/trigger/editAppealAfterSubmit)'
+
+  // For Task Event Start and Completion, when user intiates an event from the tasks tab
+  // we need a way to identify the task chosen by the user for further processing
+  // The Next steps link is configured and provided by the backend services
+  // So, instead of modifying the url, we decided that the safest approach is to
+  // append task id as a querystring.
+
+  const taskDescription = task.description;
```

Comment:
do we need to check if task is null
`if(task && task.description) {...`

Reply:
Thank you @udaydenduluri33, I have applied the changes.